.cbmc-batch/include/evp_utils.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#ifndef EVP_UTILS_H
+#define EVP_UTILS_H
+
+#include <openssl/evp.h>
+
+bool evp_pkey_is_valid(EVP_PKEY *pkey);
+
+EVP_PKEY *evp_pkey_nondet_alloc();
+
+int evp_pkey_get_reference_count(EVP_PKEY *pkey);
+
+void evp_pkey_set0_ec_key(EVP_PKEY *pkey, EC_KEY *ec);
```

Comment:
OpenSSL convention of get0 means get member without incrementing reference count.  Lucas uses set0 to mean set the member without incrementing reference count.

Reply:
Unconditional free means freeing regardless of reference count.  Shallow free means there is no reference count for the struct, but there are members with reference count so free everything but them because they will be unconditionally freed elsewhere.